dynup:
  # Engine
  engine_rate: 240
  foot_distance: 0.2
  trunk_x_final: -0.025
  hand_walkready_pitch: -60

  # Back
  leg_min_length_back: 0.22
  trunk_x_back: 0.11
  hands_behind_back_x: 0.15
  hands_behind_back_z: 0.15
  trunk_height_back_1: 0.2
  trunk_height_back_2: 0.26
  com_shift_1: 0.1
  com_shift_2: 0.1
  foot_angle: 60
  trunk_overshoot_angle_back: 0
  time_legs_close: 1
  time_foot_ground_back: 2
  time_full_squat_hands: 4
  time_full_squat_legs: 2
  wait_in_squat_back: 1

  # Front
  arm_side_offset: 0.12
  leg_min_length_front: 0.25
  trunk_x_front: 0.0
  max_leg_angle: 70
  trunk_overshoot_angle_front: 10
  hands_pitch: -30
  time_hands_side: 1
  time_hands_rotate: 1
  time_foot_close: 1
  time_hands_front: 1
  time_foot_ground_front: 1 # do at same time as time_hands_front
  time_torso_45: 1
  time_to_squat: 1
  wait_in_squat_front: 1

  # Rise
  rise_time: 1
  trunk_height: 0.4
  trunk_pitch: 0

  # Descend
  descend_time: 0.25

  # Stabilier
  stabilizing: False
  minimal_displacement: False

  # Visualizer
  spline_smoothness: 100
  display_debug: True

  pid_trunk_roll:
    p: 0.76
    i: 2.0
    d: 0.02977
    i_clamp: 0
    i_clamp_min: 0
    i_clamp_max: 0
    antiwindup: True
    publish_state: True
  pid_trunk_pitch:
    p: 0.42
    i: 1.68
    d: 0.02625
    i_clamp: 0
    i_clamp_min: 0
    i_clamp_max: 0
    antiwindup: True
    publish_state: True